### ISSUE TITLE
When users clone(), automatically create streams with custom highWaterMark

### ISSUE THREAD

**Issue by bitinn** [bitinn/node-fetch#386]
Instead of changing [default highWaterMark](https://nodejs.org/api/stream.html#stream_implementing_a_readable_stream) of `res.body` (see #151), perhaps setting them during `clone()` would be a better idea. AFAIK, users mostly run into backpressure when they clone() and consume a large response while leaving the other stream idle.

This will likely solve many trivial issues.

**Comment by DevBrent** [comment]
The https://github.com/mikeal/r2 library is particularly affected. All of their short-hand single-promise resolvers e.g. await r2(url).json, await r2(url).text are affected because they clone() every response before trying to use the native json/text resolvers.

**Comment by bitinn** [comment]
@DevBrent PR is welcomed, should be a simple change, but we do want to see a good test to verify it's working as intended.

To truly solve the highWaterMark limit you need a dynamically sizing stream, I am not sure it's possible, but I haven't looked at latest nodejs development.

**Comment by DevBrent** [comment]
@bitinn I'm not even sure I like what seems is the only solution. On one hand, no reader should be prevented from reading the entire stream if they read faster than another reader. On the other, a sufficiently fast reader or in situations like with r2, dynamically sizing the stream will cause the entire stream to be stored in-memory which is not scalable.

**Comment by nikdo** [comment]
I've done some initial research about `highWaterMark` behavior when one of the streams does not have a consumer:

```js
const highWaterMark = 5

// push byte-sized chunks to reach +1 over buffer limit to trigger backpressure
const input = new RandomBytes({
  // Transform stream has 2 buffers it absorbes twice the buffer before backpressuring
  // https://nodejs.org/api/stream.html#stream_buffering
  pushes: highWaterMark * 2 + 1,
  verbose: false
})

// highWaterMark limit is problem only for the stream without consumer
let p1 = new PassThrough({ highWaterMark })
let p2 = new PassThrough()

// cannot hoook onto p1 data because that would add consumer
p2.on('data', data => {
  logBuffer(p1, 'p1')
})

input.pipe(p1) // no pipe for you
input.pipe(p2).pipe(fs.createWriteStream('/dev/null'))
```

[Full code](https://gist.github.com/nikdo/2b15487efbb50a6e102fff3ba4b14250)

Because [`PassThrough`](https://github.com/bitinn/node-fetch/blob/master/src/body.js#L377) is a Transform stream, it has two buffers. That means the backpressure occurs when there is `16 * 1024 * 2 + 1` on the input.

Now I'll investigate how to properly test pipe that does not emit `end` event leading to unresolved promise. Not really _good first issue_ when you want to see a good test to verify it's working as intended. :) Any help appreciated.

**Comment by xxczaki** [comment]
Fixed by #671 :tada:
